src/Bridge/Doctrine/Orm/Filter/SearchFilter.php

```diff
@@ -101,8 +101,15 @@ public function apply(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $q
                 $parentAlias = $alias;
 
                 foreach ($propertyParts['associations'] as $association) {
-                    $alias = $queryNameGenerator->generateJoinAlias($association);
-                    $queryBuilder->join(sprintf('%s.%s', $parentAlias, $association), $alias);
+                    $join = $this->hasJoin($queryBuilder, $alias, $association);
+
+                    if (false === $join) {
```

Comment:
Can't all the following block moved in the private function? It looks duplicated.